src/java/org/apache/cassandra/index/sai/plan/QueryView.java

```diff
@@ -111,201 +102,72 @@ static class Builder
         static class MissingIndexException extends RuntimeException
         {
             final IndexContext context;
-            final String dataObjectName;
 
-            private MissingIndexException(IndexContext context, String dataObjectName)
+            private MissingIndexException(IndexContext context)
             {
                 super();
                 this.context = context;
-                this.dataObjectName = dataObjectName;
-            }
-
-            public static MissingIndexException forSSTable(IndexContext context, Descriptor descriptor)
-            {
-                return new MissingIndexException(context, "sstable " + descriptor);
-            }
-
-            public static MissingIndexException forMemtable(IndexContext context, Memtable memtable)
-            {
-                return new MissingIndexException(context, "memtable " + memtable);
             }
 
             @Override
             public String getMessage()
             {
-                return "Index " + context.getIndexName() + " not found for " + dataObjectName;
+                return context.isDropped() ? "Index " + context.getIndexName() + " was dropped."
```

Comment:
An Exception that holds references to stateful objects smells a little bit to me. Because exception often bubble up and escape from the original context and referenced objects may be not valid anymore.

 What about building the message in the constructor and not reference

Reply:
Fixed